devito/types/sparse.py

```diff
@@ -579,15 +584,15 @@ def _coordinate_indices(self):
                       for c, o, i in zip(self._coordinate_symbols, self.grid.origin,
                                          indices[:self.grid.dim])])
 
-    @cached_property
-    def _coordinate_bases(self):
+    def _coordinate_bases(self, field_offset):
         """Symbol for the base coordinates of the reference grid point."""
         indices = self.grid.dimensions
         return tuple([cast_mapper[self.dtype](c - o - idx * i.spacing)
-                      for c, o, idx, i in zip(self._coordinate_symbols,
-                                              self.grid.origin,
-                                              self._coordinate_indices,
-                                              indices[:self.grid.dim])])
+                      for c, o, idx, i, of in zip(self._coordinate_symbols,
+                                                  self.grid.origin,
+                                                  self._coordinate_indices,
+                                                  indices[:self.grid.dim],
```

Comment:
does this `:self.grid.dim` still make sense? it's quite horrible

Reply:
I think so but good to think of for another time